oktapam/data_source_database.go

```diff
@@ -0,0 +1,108 @@
+package oktapam
+
+import (
+	"context"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/okta/terraform-provider-oktapam/oktapam/client"
+	"github.com/okta/terraform-provider-oktapam/oktapam/client/wrappers"
+	"github.com/okta/terraform-provider-oktapam/oktapam/constants/attributes"
+	"github.com/okta/terraform-provider-oktapam/oktapam/constants/descriptions"
+)
+
+func dataSourceDatabase() *schema.Resource {
+	return &schema.Resource{
+		Description: descriptions.SourceDatabase,
+		ReadContext: dataSourceDatabaseFetch,
+		Schema: map[string]*schema.Schema{
+			attributes.ID: {
+				Type:     schema.TypeString,
+				Required: true,
+				// Description is autogenerated
+			},
+			attributes.ResourceGroup: {
+				Type:        schema.TypeString,
+				Required:    true,
+				Description: descriptions.ResourceGroupID,
+			},
+			attributes.Project: {
+				Type:        schema.TypeString,
+				Required:    true,
+				Description: descriptions.ProjectID,
+			},
+			attributes.CanonicalName: {
+				Type:        schema.TypeString,
+				Optional:    true,
+				Description: descriptions.CanonicalName,
+			},
+			attributes.DatabaseType: {
+				Type:        schema.TypeString,
+				Optional:    true,
+				Description: descriptions.DatabaseType,
+			},
+			attributes.RecipeBook: {
+				Type:        schema.TypeString,
+				Optional:    true,
+				Description: descriptions.RecipeBookID,
+			},
+			attributes.ManagementConnectionDetails: {
+				Type:        schema.TypeList,
+				Optional:    true,
+				MinItems:    1,
+				MaxItems:    1,
+				Description: descriptions.ManagementConnectionDetails,
+				Elem:        dbConnectionTypes,
+			},
+			attributes.ManagementGatewaySelector: {
+				Type:        schema.TypeMap,
+				Optional:    true,
+				Description: descriptions.ManagementGatewaySelector,
+				Elem: &schema.Schema{
+					Type: schema.TypeString,
+				},
+			},
+			attributes.ManagementGatewaySelectorID: {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: descriptions.ManagementGatewaySelectorID,
+			},
+			attributes.CreatedAt: {
```

Comment:
Usually we are not keeping create_at/update_at fields in datasource or resources as Sajan mentioned in other comment. Also, don't think you are setting these while converting ToResourceMap.

Reply:
removed